extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java

```diff
@@ -195,9 +206,120 @@ public void build(ApplicationInfoBuildItem applicationInfo,
+     * @param name The name of the resource to accept the configuration
+     * @param config The {@link PlatformConfiguration} instance
+     */
+    private void applyConfig(Session session, String target, String name, PlatformConfiguration config) {
+        //Labels
+        config.getLabels().entrySet().stream().forEach(e -> {
+            session.resources().decorate(target, new AddLabelDecorator(new Label(e.getKey(), e.getValue())));
+        });
+
+        //Annotations
+        config.getAnnotations().entrySet().stream().forEach(e -> {
+            session.resources().decorate(target, new AddAnnotationDecorator(new Annotation(e.getKey(), e.getValue())));
+        });
+
+        //EnvVars
+        config.getEnvVars().entrySet().stream().forEach(e -> {
+            session.resources().decorate(target, new AddEnvVarDecorator(EnvConverter.convert(e)));
+        });
+
+        config.getWorkingDir().ifPresent(w -> {
+            session.resources().decorate(target, new ApplyWorkingDirDecorator(name, DEPLOY));
+        });
+
+        config.getCommand().ifPresent(c -> {
+            session.resources().decorate(target,
+                    new ApplyCommandDecorator(name, c.toArray(new String[c.size()])));
+        });
+
+        config.getArguments().ifPresent(a -> {
+            session.resources().decorate(target, new ApplyArgsDecorator(name, a.toArray(new String[a.size()])));
+        });
+
+        config.getServiceAccount().ifPresent(s -> {
+            session.resources().decorate(target, new ApplyServiceAccountNamedDecorator(name, s));
+        });
+
+        //Image Pull
+        session.resources().decorate(new ApplyImagePullPolicyDecorator(config.getImagePullPolicy()));
+        config.getImagePullSecrets().ifPresent(l -> {
+            l.forEach(s -> session.resources().decorate(target, new AddImagePullSecretDecorator(name, s)));
+        });
+
+        //Probes
+        config.getLivenessProbe().ifPresent(p -> {
+            session.resources().decorate(target, new AddLivenessProbeDecorator(name, ProbeConverter.convert(p)));
+        });
+
+        config.getReadinessProbe().ifPresent(p -> {
+            session.resources().decorate(target,
+                    new AddReadinessProbeDecorator(name, ProbeConverter.convert(p)));
+        });
+
+        // Mounts and Volumes
+        config.getMounts().entrySet().forEach(e -> {
+            session.resources().decorate(target, new AddMountDecorator(MountConverter.convert(e)));
+        });
+
+        config.getSecretVolumes().entrySet().forEach(e -> {
+            session.resources().decorate(target, new AddSecretVolumeDecorator(SecretVolumeConverter.convert(e)));
+        });
+
+        config.getConfigMapVolumes().entrySet().forEach(e -> {
+            session.resources().decorate(target, new AddConfigMapVolumeDecorator(ConfigMapVolumeConverter.convert(e)));
+        });
+
+        //config.getGitRepoVolumes().entrySet().forEach(e -> {
```

Comment:
What's the deal with this one?

Reply:
There is no such decorator. So, until its implemented we cant use it.

Reply:
I'll remove it.

Reply:
Thanks